pkg/stacks/steps.go

```diff
@@ -78,16 +78,29 @@ func appStep(sp StackPackager) walker.Step {
 	}
 }
 
-// behaviorStep unmarshals stack.yaml bytes
-func behaviorStep(sp StackPackager) walker.Step {
+// stackConfigStep unmarshals stack.yaml bytes
+func stackConfigStep(sp StackPackager) walker.Step {
 	return func(path string, b []byte) error {
 		// unstructured is used so that 'omitempty' will be respected after unmarshaling and marshaling
 		config := unstructured.Unstructured{}
 		if err := yaml.Unmarshal(b, &config); err != nil {
 			return errors.Wrap(err, fmt.Sprintf("invalid stack configuration at %q", path))
 		}
 
-		sp.SetBehavior(config)
+		sp.SetStackConfig(config)
+		return nil
+	}
+}
+
+// behaviorStep unmarshals stack.yaml bytes
+func behaviorStep(sp StackPackager) walker.Step {
+	return func(path string, b []byte) error {
+		behavior := v1alpha1.StackDefinitionBehavior{}
+		if err := yaml.Unmarshal(b, &behavior); err != nil {
```

Comment:
If the file is missing, this step will not be called. The path is only for context information (used by other steps where the filename adds details about what CRD is being described).

If the file were empty (or providing alternate yaml documents), the behavior object would be default/empty, but it will still be known.  We could add validation in `behaviorStep` to prevent illogical configurations.

If the contents were not yaml conformant and error would be returned.